src/main/java/FlyChat/Core/Storage.java

```diff
@@ -0,0 +1,68 @@
+package FlyChat.Core;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
+/**
+ * Contains operations related to reading and writing of the save file.
+ */
+public class Storage {
+    private File saveFile;
+
+    /**
+     * Finds the save file specified in the input filePath. If no save file is found, creates a
+     * new save file.
+     */
+    public void findSaveFile(String filePath) {
+        try {
+            saveFile = new File(filePath);
+            saveFile.createNewFile();
+        } catch (IOException e) {
+            System.out.println("Save file could not be created. Info will not be saved.");
+        }
+    }
+
+    /**
+     * Reads the save file and saves all tasks present in the save file to the task list.
+     */
+    public void loadSaveFile(TaskList loadTarget) {
+        try {
+            Scanner saveReader = new Scanner(saveFile);
+            while (saveReader.hasNextLine()) {
+                loadTarget.addTaskFromFile(saveReader.nextLine());
+            }
+            saveReader.close();
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+            saveFile.delete();
+            try {
+                saveFile.createNewFile();
+            } catch (IOException f) {
+                System.out.println("Save file could not be created. Info will not be saved.");
+            }
+        }
+    }
+
+    /**
+     * Overwrites the current save file with all tasks from the current task list.
+     *
+     * @param source String to be written into the save file.
+     */
+    public void writeToSave(String source) {
+        //Replaces old file with a new file with updated contents
+        try {
+            File tmp = File.createTempFile("tmp", "");
+            BufferedWriter writer = new BufferedWriter(new FileWriter(tmp));
+            writer.write(source);
+            writer.close();
+            if(saveFile.delete()) {
+                tmp.renameTo(saveFile);
+            }
+        } catch (IOException e) {
+            System.out.println("Unable to write to file. Information may not be saved for the next session.");
```

Comment:
It's great that you gave this error message, I think the user will benefit knowing that the session won't be saved!